packages/jest-cli/src/runGlobalHook.js

```diff
@@ -27,24 +33,42 @@ export default ({
   }
 
   if (globalModulePaths.size > 0) {
-    return Promise.all(
-      Array.from(globalModulePaths).map(async modulePath => {
-        if (!modulePath) {
-          return null;
-        }
-
-        // $FlowFixMe
-        const globalModule = require(modulePath);
-
-        if (typeof globalModule !== 'function') {
-          throw new TypeError(
-            `${moduleName} file must export a function at ${modulePath}`,
-          );
-        }
-
-        return globalModule(globalConfig);
-      }),
-    );
+    return pEachSeries(Array.from(globalModulePaths), async modulePath => {
```

Comment:
there's a subtle difference here - previously they ran in parallel while they'll now run sequentially. This is purpose so that the require hook is only applied once at a time (since)

Reply:
This sounds like it may be the cause of https://github.com/facebook/jest/issues/7710 actually... Maybe 🤔

Reply:
the cause is on line 54 - we try to load your transformer in order to transform the `globalSetup` file, but we need the transformer to transform the transformer first (whew 😅). Line 57 will break the cycle if you make sure to ignore your transformer (via `transformIgnorePatterns`, check the updated integrations tests in this diff)

Reply:
We should probably add an error if a transformer is configured with a pattern that will match a `globalSetup` or `globalTeardown`. Or if that's difficult, special case the "transformer must export..." error inside of the transform here

Reply:
Wait, if a transformer has a pattern that will hit _itself_, _and_ you have either `globalSetup` or `globalTeardown`.

Reply:
#7752